appinventor/appengine/src/com/google/appinventor/server/LoginServlet.java

```diff
@@ -296,39 +296,39 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws I
     if (page.equals("sendlink")) {
       String email = params.get("email");
       if (email == null) {
-        fail(req, resp, "No Email Address Provided");
+        fail(req, resp, locale, "No Email Address Provided");
         return;
       }
       // Send email here, for now we put it in the error string and redirect
       PWData pwData = storageIo.createPWData(email);
       if (pwData == null) {
-        fail(req, resp, "Internal Error");
+        fail(req, resp, locale, "Internal Error");
         return;
       }
       String link = trimPage(req) + pwData.id + "/setpw";
       sendmail(email, link, locale);
-      resp.sendRedirect("/login/linksent/");
+      resp.sendRedirect("/login/linksent/?locale=" + locale);
```

Comment:
`locale` isn't used in that page, so it's not really necessary to feed it the locale.